here-naksha-lib-ext-manager/src/main/java/com/here/naksha/lib/extmanager/ExtensionCache.java

```diff
@@ -83,24 +82,25 @@ protected void buildExtensionCache(ExtensionConfig extensionConfig) {
       publishIntoCache(result, extensionConfig);
     });
 
-    // Removing existing extension which has been removed from the configuration
-    List<String> extIds =
-        extensionConfig.getExtensions().stream().map(Extension::getId).toList();
+    // Removing existing extension which has been removed from the configuration.
+    List<String> extIds = extensionConfig.getExtensions().stream()
+        .map(extension -> extension.getEnv() + ":" + extension.getId())
+        .toList();
 
     for (String key : loaderCache.keySet()) {
       if (!extIds.contains(key)) {
-        loaderCache.remove(key);
-        PluginCache.removeExtensionCache(key);
-        logger.info("Extension {} removed from cache.", key);
-      }
+        removeExtensionFromCache(key);
+        }
     }
     logger.info("Extension cache size " + loaderCache.size());
   }
 
   private void publishIntoCache(KVPair<Extension, File> result, ExtensionConfig extensionConfig) {
     if (result != null && result.getValue() != null) {
       final Extension extension = result.getKey();
+      final String extensionIdWthEnv = extension.getEnv() + ":" + extension.getId();
       final File jarFile = result.getValue();
+      IExtensionInit instance = null;
```

Comment:
Suggest renaming "instance" variable to "initObj" , to avoid confusion between Extension instance and init class instance

Reply:
Done.